internal/alpini/netty4/alpini-netty4-base/src/main/java/com/linkedin/alpini/netty4/ssl/SslInitializer.java

```diff
@@ -577,9 +604,19 @@ public void operationComplete(Future<String> future) {
         }
 
         private void resolved(Future<? super Void> future) {
+          // Step 3.5: This function is called after DNS resolution completes (if _resolveClient enabled); next step is
+          // to replace itself with SSL handler. If _resolveClient not enabled, SslDetect would be replaced
+          // with SSL handler at the decode function directly, before it has a chance to reach this function.
           // This is executed on the channel executor thread
           assert _channelHandlerContext.channel().eventLoop().inEventLoop();
 
+          Long dnsStartTime = (Long) _channelHandlerContext.channel().attr(DNS_START_TS).get();
+          long dnsLatencyMs = TimeUnit.NANOSECONDS.toMillis(Time.nanoTime() - dnsStartTime);
+          LOG.log(
```

Comment:
can this blowup during new connection setup and slow dns ? Do we need to rate limit it?

Reply:
This one is probably okay, since it's only one line per request only when DNS resolution is very slow.

I added an error log with stacktrace when SSL handshake fails, which is concerning. Planning to remove the stacktrace after finishing all the reproduction tests. Will probably keep this one as it is. Notice that there is a log with latency for every successful handshake too.